onnxruntime/core/language_interop_ops/torch/torch_proxy.h

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#include <Python.h>
+#include <mutex>
+
+#ifndef SHARED_PROVIDER
+#include "core/platform/env.h"
+#include "core/framework/ml_value.h"
+#include "core/framework/op_kernel_context_internal.h"
+#endif
+
+#include "core/language_interop_ops/torch/object_pointer.h"
+
+namespace onnxruntime {
+namespace language_interop_ops {
+namespace torch {
+
+// PyObject RAII wrapper
+using PythonObjectPtr = ObjectPointer<PyObject>;
+template class ObjectPointer<PyObject>;
+
+class TorchProxy {
+ public:
+  static TorchProxy& GetInstance() {
+    static TorchProxy proxy;
+    return proxy;
+  };
+
+  void Forward(
+      void* callback,
+      const std::vector<int64_t>& requires_grads,
+      const std::vector<OrtValue>& tensor_args,
+      const std::vector<int64_t>& tensor_indices,
+      const std::vector<void*>& obj_args,
```

Comment:
Can we add a comment here for the use of void* (here and on diff_ctx below)?  This is really a vector of PyObject*, and so is the aim to avoid Python-specific implementation "polluting" the rest of the tree / adding dependencies?

Reply:
I recalled this is a historical reason, when we did not want polute the head file with <python.h.  But later on we find it is okay because torch_proxy.h/.cc anyway will depends on Python lib. So I think we can change it to PyObject*. I will do that.

Reply:
Add comments for torch_proxy.h and orttraining/orttraining/training_ops/cpu/torch/torch_custom_function_kernel_base.h. 

And yes, we don't want to bring extra Python.h dependency for those headers' consumers.